src/transformers/models/opt/modeling_opt.py

```diff
@@ -724,7 +745,13 @@ def forward(
                 attention_mask, input_shape, inputs_embeds, past_key_values_length
             )
 
-        pos_embeds = self.embed_positions(attention_mask, past_key_values_length)
+        if position_ids is None:
+            position_ids = torch.cumsum(attention_mask, dim=1)
+            position_ids = (position_ids * attention_mask - 1).long()
+            # cut positions if `past_key_values_length` is > 0
+            position_ids = position_ids[:, past_key_values_length:]
```

Comment:
only thing against this is that it's not compile compatible (dynamic slicing on input dependent) but okay!

Reply:
is there a way to make it compile compatible?

Reply:
creating a tensor with `torch.arange(past_key_values_length, full_length)` but we would need to test!

Reply:
Ok, do you want me to try? and how do you test if it is compile compatiable?

Reply:
We have a slow test for this: https://github.com/huggingface/transformers/blob/fe496262c119e4366b2f319567bb561cab445dc1/tests/test_modeling_common.py#L4986 to make sure it runs!

Reply:
how can i make the ```torch.arange(past_key_values_length, full_length)``` work with multiplying attention_mask (similar to ```position_ids * attention_mask```. it will need dynamic slicing as well?
does dynamic slicing compile if it does not under ```if position_ids is None```?

Reply:
@ArthurZucker after looking at the test, I think it will not work for opt. the test uses static cache and opt doesn't support static cache. So making compile work for OTP will need further work

Reply:
Ahhh right, I thought #32617 had touched OPT, but not yet! 
Then no worries

Reply:
OK